eventstreamrpc/include/aws/eventstreamrpc/EventStreamClient.h

```diff
@@ -190,44 +190,77 @@ namespace Aws
             virtual void OnContinuationClosed() = 0;
         };
 
-        enum EventStreamErrors
+        enum EventStreamRpcError
         {
             /* If error messages are added to `aws_event_stream_errors`, this will need to be updated. */
-            AWS_ERROR_EVENT_STREAM_RPC_UNKNOWN_PROTOCOL_MESSAGE = AWS_ERROR_EVENT_STREAM_RPC_STREAM_NOT_ACTIVATED + 1,
-            AWS_ERROR_EVENT_STREAM_RPC_UNMAPPED_DATA,
-            AWS_ERROR_EVENT_STREAM_RPC_UNSUPPORTED_CONTENT_TYPE,
-            AWS_ERROR_EVENT_STREAM_RPC_STREAM_CLOSED_ERROR
+            EVENT_STREAM_RPC_SUCCESS = 0,
+            EVENT_STREAM_RPC_NULL_PARAMETER,
+            EVENT_STREAM_RPC_INITIALIZATION_ERROR,
+            EVENT_STREAM_RPC_CONNECTION_CLOSED_BEFORE_CONNACK,
+            EVENT_STREAM_RPC_UNKNOWN_PROTOCOL_MESSAGE,
+            EVENT_STREAM_RPC_UNMAPPED_DATA,
+            EVENT_STREAM_RPC_UNSUPPORTED_CONTENT_TYPE,
+            EVENT_STREAM_RPC_STREAM_CLOSED_ERROR,
+            EVENT_STREAM_RPC_UNEXPECTED_ERROR,
+            EVENT_STREAM_RPC_CRT_ERROR
+        };
+
+        struct EventStreamRpcStatus
+        {
+            EventStreamRpcError baseStatus;
+            int crtError;
+        };
+
+        template <class T> class ProtectedPromise
```

Comment:
Note that this type will never be public facing and is just a helper class. Ultimately, we still return an `std::future`.